client/src/pages/Bookings/Bookings.tsx

```diff
@@ -6,45 +6,25 @@ import PageContainer from '../../components/PageContainer/PageContainer';
 import NextBooking from '../../components/NextBookings/NextBooking';
 import { Booking } from '../../interface/Booking';
 import BookingList from '../../components/BookingList/BookingList';
-import { getBookings } from '../../helpers/APICalls/requests';
-import { useState, useEffect, useCallback } from 'react';
 import { useSnackBar } from '../../context/useSnackbarContext';
+import { useBookings } from '../../context/useBookingsContext';
 
 const Bookings = () => {
   const { loggedInUser } = useAuth();
   const history = useHistory();
-  const [bookings, setBookings] = useState<Booking[]>([]);
+  const { bookings } = useBookings();
   const { updateSnackBarMessage } = useSnackBar();
-
-  const fetchBookings = useCallback(async () => {
-    try {
-      const response = await getBookings();
-      const { success } = response;
-      if (success) {
-        success.requests.forEach((request) => {
-          const { start, end } = request;
-          request.start = new Date(start);
-          request.end = new Date(end);
-        });
-        setBookings(success.requests);
-      } else {
-        updateSnackBarMessage('Error fetching bookings from server');
-      }
-    } catch (e) {
-      const { message } = e as Error;
-      updateSnackBarMessage(message);
-    }
-  }, [updateSnackBarMessage]);
-
-  useEffect(() => {
-    fetchBookings();
-  }, [fetchBookings]);
-
-  if (loggedInUser === undefined || !bookings) return <CircularProgress />;
+  if (loggedInUser === undefined || bookings === undefined) return <CircularProgress />;
   if (!loggedInUser) {
     history.push('/login');
     return <CircularProgress />;
   }
+
+  if (!bookings) {
+    updateSnackBarMessage('Error fetching bookings');
+    return <></>;
+  }
```

Comment:
Optionally, I'd like to clean these checks up a little bit. `loggedInUser` logic should be handled by `ProtectedRoute`s (#28). We can focus on falling back on a spinner while we load bookings (e.g. maybe use `isLoading` in the context instead of comparing against `null/undefined`?)